### PR TITLE
Feat: Authentication

### PR DESCRIPTION
**Pull request by jonas-wedemeyer**

### CODE REVIEW

**Review by miquintero — APPROVED** [review]
Looks good! Thank you for updating the User model!!!

File: controller/users.js [hunk 1]
```diff
@@ -1,14 +1,78 @@
...
+    } else {
+      if (userData.password) {
+        const hash = await bcrypt.hash(userData.password, 10);
+        userData.password = hash;
+      }
+      user = await User.create({ ...userData });
+      user = filterProps(user.dataValues, [
+        'id',
+        'picture',
+        'password',
```
Comment (tonyghiani) [inline controller/users.js:26]: Don't send back to the client the password, this expose important information in the client.

File: controller/users.js [hunk 2]
```diff
@@ -1,14 +1,78 @@
...
+    const [email, password] = authHeader;
+    let user = await User.findOne({ where: { email } });
+
+    if (user) {
+      const pwMatch = await bcrypt.compare(password, user.password);
+      if (pwMatch) {
+        user = filterProps(user.dataValues, [
+          'id',
+          'picture',
+          'password',
```
Comment (tonyghiani) [inline controller/users.js:55]: Same here.

File: controller/users.js (hunk 1 above)
Comment (jonas-wedemeyer) [reply controller/users.js:26]: The elements in the array state, which entries should be removed from the object. Hence password is currently not send back to the client. Let's discuss tomorrow.

File: controller/users.js (hunk 2 above)
Comment (jonas-wedemeyer) [reply controller/users.js:55]: s. above

File: controller/users.js (hunk 1 above)
Comment (tonyghiani) [reply controller/users.js:26]: I see, the name was misunderstanding since i thought was keeping those props. Ok cool, kn this case it works :)

**Pull request merged**